### PR TITLE
add recipe for TDLib

### PR DESCRIPTION
**Pull request by aplavin**
telegram API library

### CODE REVIEW

**Comment by aplavin** [comment]
This is the lib I asked about on slack yesterday. Turns out there is a flag that makes it compatible with old glibc.

File: T/TDLib/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,46 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder, Pkg
+
+name = "tdlib"
```
Comment (giordano) [inline T/TDLib/build_tarballs.jl:5]: This _must_ match exactly the directory name, including the case
```suggestion
name = "TDLib"
```

File: T/TDLib/build_tarballs.jl [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+mkdir build
+cd build
+cmake -DCMAKE_BUILD_TYPE=Release -DZLIB_LIBRARY=$libdir -DZLIB_INCLUDE_DIR=$includedir ..
+for f in $(grep -l $(find . -name link.txt) -e 'opt/bin/c++')
+do
+    # without this flag it cannot find libz
+    sed -i ' 1 s/$/ -lz/' $f
+done
+cmake --build . -j${nproc}
+cp libtdjson.so $libdir
```
Comment (giordano) [inline T/TDLib/build_tarballs.jl:28]: I assume this needs to be
```suggestion
cp libtdjson.${dlext} $libdir/libtdjson.${dlext}
```

Comment (aplavin) [reply T/TDLib/build_tarballs.jl:28]: Building for windows errors anyways.

File: T/TDLib/build_tarballs.jl [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+"""
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+platforms = filter(p -> Sys.islinux(p) && (libc(p) == "glibc" || arch(p) == "x86_64"), supported_platforms())
```
Comment (giordano) [inline T/TDLib/build_tarballs.jl:33]: The problem is that the build system is trying to run a foreign executable.  I have a small hope we can avoid that, but I don't have the time to look into it this week

File: T/TDLib/build_tarballs.jl [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    # without this flag it cannot find libz
+    sed -i ' 1 s/$/ -lz/' $f
+done
+cmake --build . -j${nproc}
+cp libtdjson.so $libdir
+"""
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+platforms = filter(p -> Sys.islinux(p) && (libc(p) == "glibc" && arch(p) == "i686" || arch(p) == "x86_64"), supported_platforms())
```
Comment (giordano) [inline T/TDLib/build_tarballs.jl:33]: This is simpler
```suggestion
platforms = filter(p -> Sys.islinux(p) && (libc(p) == "glibc" && proc_family(p) == "intel"), supported_platforms())
```

Comment (aplavin) [reply T/TDLib/build_tarballs.jl:33]: it's "x64 with any libc" or "i686 with glibc"

Comment (giordano) [reply T/TDLib/build_tarballs.jl:33]: I know, I was just pointing out that using `proc_family(p)` gives the same condition as yours, but fewer keystrokes

File: T/TDLib/build_tarballs.jl (hunk 3 above)
Comment (giordano) [reply T/TDLib/build_tarballs.jl]: Just to understand, do you want to try to build for other platforms or happy to support only a few of them?

Comment (aplavin) [reply T/TDLib/build_tarballs.jl]: I'm not familiar with such details unfortunately. Custom flags in this script come from just googling specific compilation errors and applying fixes I found somewhere.

File: T/TDLib/build_tarballs.jl (hunk 4 above)
Comment (aplavin) [reply T/TDLib/build_tarballs.jl:33]: Cannot reproduce it locally, says `ERROR: UndefVarError: proc_family not defined`. But looks like your variant won't select x64 musl.

Comment (giordano) [reply T/TDLib/build_tarballs.jl:33]: Oh, I see, I misread your condition, I thought you were skipping that platform on purpose

**Pull request merged**